controller/handler/https.go

```diff
@@ -159,14 +160,14 @@ func (h HTTPS) Update(k store.K8s, cfg *config.ControllerCfg, api api.HAProxyCli
 	_, errFtSSL := api.FrontendGet(cfg.FrontSSL)
 	if cfg.SSLPassthrough {
 		if errFtSSL != nil {
-			logger.Error(h.enableSSLPassthrough(cfg, api))
+			logger.Error(h.enableSSLPassthrough(k, cfg, api))
```

Comment:
Maybe no need to send the whole store but only the tls-alpn value at this stage 
Or you can even have a private field in handler struct that you can set in the main `Update` method:
```
h.alpn = annotations.GetValue("tls-alpn", k.ConfigMaps.Main.Annotations)
```
Then later you can use it when calling `FrontendEnableSSLOffload`
```
api.FrontendEnableSSLOffload(cfg.FrontHTTPS, h.CertDir, h.alpn)
```

Reply:
Yup, that's a lot simpler! I will add a field to the HTTPS struct and set the value at the beginning of the Update method